src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +25,32 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        Set<Character> set = new TreeSet<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            while (reader.ready()) {
+                String charAsString = ((char) reader.read() + "")
+                        .replaceAll("[_\\W\\d]", "").toLowerCase();
+                if (!charAsString.equals("")) {
+                    set.add(charAsString.toCharArray()[0]);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            throw new FileNotFoundException();
```

Comment:
@vztot Чому б не прокидувати так:
```suggestion
            throw new FileNotFoundException("Some message you think will be suitable", e);
```